### PR TITLE
Merge performance report, Class Performance Imbalance and Train test Overfit into one check

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/checks/performance/performance_report.py [hunk 1]
```diff
@@ -9,20 +9,24 @@
...
+from deepchecks import CheckResult, Dataset, TrainTestBaseCheck, ConditionResult, ModelComparisonBaseCheck
+from deepchecks.utils.metrics import MULTICLASS_SCORERS_NON_AVERAGE, get_scorers_list, initialize_multi_scorers, ModelType, task_type_check
 from deepchecks.utils.validation import validate_model
 
 
 __all__ = ['PerformanceReport', 'MultiModelPerformanceReport']
 
 
-class PerformanceReport(SingleDatasetBaseCheck):
+PR = TypeVar('PR', bound='PerformanceReport')
```
[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/checks/performance/performance_report.py:28]: impossible

File: deepchecks/checks/performance/performance_report.py [hunk 2]
```diff
@@ -45,41 +70,186 @@ def run(self, dataset, model=None) -> CheckResult:
...
+        scorers = get_scorers_list(model, test_dataset, self.alternative_scorers, multiclass_avg=False)
+        datasets = {'Train': train_dataset, 'Test': test_dataset}
+        if task_type in [ModelType.BINARY, ModelType.MULTICLASS]:
+            x = ['Class', 'Dataset']
+            results = []
+            for dataset_name, dataset in datasets.items():
+                for scorer in scorers:
+                    score_result = scorer(model, dataset)
+                    # Multiclass scorers return numpy array of result per class
+                    for class_i, value in enumerate(score_result):
```
Comment (matanper) [inline deepchecks/checks/performance/performance_report.py:102]: what about binary scorers?

Comment (matanper) [reply deepchecks/checks/performance/performance_report.py:102]: This won't work if the labels are not consecutive (for example the labels are 1,2,6)
use this: 
```
for class_score, class_value in zip(score_result, test_dataset.classes):
```

**Review by matanper — CHANGES_REQUESTED** [review]

**Review by matanper — APPROVED** [review]
